### PR TITLE
improved notices.show_notices

### PR DESCRIPTION
**Pull request by CantoroMC**
- Prefer use of lua functions instead of vim.cmd
- Added buffer and window options
- Removed extra empty lines

### CODE REVIEW

File: lua/lualine/utils/notices.lua [hunk 1]
```diff
@@ -36,12 +36,27 @@ function M.notice_message_startup()
...
-  vim.cmd [[
-  :silent! new
-  :silent! setl ft=markdown bt=nofile nobuflisted bh=wipe
-  :silent! nnoremap <silent><buffer> q <cmd>bd<cr>
-  :silent! normal ggdG
-  ]]
+  vim.cmd('silent! keepalt split')
+
+  local winid = vim.fn.win_getid()
+  local bufnr = vim.api.nvim_create_buf(false, false)
```
Comment (shadmansaleh) [inline lua/lualine/utils/notices.lua:42]: Use a scratch buffer . 2nd arg should be true.

File: lua/lualine/utils/notices.lua [hunk 2]
```diff
@@ -36,12 +36,27 @@ function M.notice_message_startup()
...
+
+  vim.wo[winid].winfixheight   = true
+  vim.wo[winid].winfixwidth    = true
+  vim.wo[winid].number         = false
+  vim.wo[winid].foldcolumn     = '0'
+  vim.wo[winid].relativenumber = false
+  vim.wo[winid].signcolumn     = 'no'
+
+  vim.bo[bufnr].filetype  = 'markdown'
+  vim.bo[bufnr].buflisted = false
```
Comment (shadmansaleh) [inline lua/lualine/utils/notices.lua:55]: I think buflisted is already set false by `nvim_create_buf` .

Also you aren't setting `buftype` and `bufhidden` options that were previously set . why?

File: lua/lualine/utils/notices.lua [hunk 3]
```diff
@@ -36,12 +36,27 @@ function M.notice_message_startup()
...
-  :silent! setl ft=markdown bt=nofile nobuflisted bh=wipe
-  :silent! nnoremap <silent><buffer> q <cmd>bd<cr>
-  :silent! normal ggdG
-  ]]
+  vim.cmd('silent! keepalt split')
+
+  local winid = vim.fn.win_getid()
+  local bufnr = vim.api.nvim_create_buf(false, false)
+
+  vim.api.nvim_set_current_buf(bufnr)
```
Comment (shadmansaleh) [inline lua/lualine/utils/notices.lua:44]: Do you need this ? Won't call to nvin_win_set_buf following line attach the buffer ?

**Comment by shadmansaleh** [comment]
I think we should use `fn.appenedbufline` instead of `fn.append` so we can specify the bufnr .

Try something like
```
vim.fn.appendbufline(bufnr, 0, notice)
```
inplace of
```
vim.fn.append(0, notice)
```

**Comment by CantoroMC** [comment]
They are both fine, the current buffer is Lua notices anyway.
If you want i change it (in this case should I add a commit on top or make it an unique commit?); otherwise you can do it while merging.

**Comment by shadmansaleh** [comment]
> They are both fine, the current buffer is Lua notices anyway.

Yeah . But if something switches current buffer while we are writting we may end up appending to different buffer . Since we're cleaning this up specifying bufnr would be a good change IMO.

> If you want i change it (in this case should I add a commit on top or make it an unique commit?);

You can add as many commits as you like I'm squash them before merge.

**Comment by shadmansaleh** [comment]
Add back the `bufhidden` and `buftype` options and we can merge this I think .

**Comment by CantoroMC** [comment]
This options are setted by: nvim_create_buf(false,true)
setlocal buftype=nofile
setlocal bufhidden=hide
setlocal noswapfile
setlocal nobuflisted

**Comment by shadmansaleh** [comment]
Ohh right part of scratch buffer . 

lgtm . Thanks for your contributions :)

**Comment by CantoroMC** [comment]
Thank you

**Pull request merged**